ConfigSpace/hyperparameters.pyx

```diff
@@ -485,21 +510,26 @@ cdef class UniformFloatHyperparameter(FloatHyperparameter):
     def _sample(self, rs: np.random, size: Optional[int]=None) -> Union[float, np.ndarray]:
         return rs.uniform(size=size)
 
-    def _transform(self, vector: Union[np.ndarray, float, int]) -> Optional[Union[np.ndarray, float, int]]:
-        if isinstance(vector, np.ndarray):
-            if np.any(np.isnan(vector)):
-                return None
-        elif np.isnan(vector):
-            return None
-        vector *= (self._upper - self._lower)
-        vector += self._lower
+    cpdef np.ndarray _transform_vector(self, np.ndarray vector):
+        if np.any(~np.isfinite(vector)):
```

Comment:
Is this strictly necessary? This function is right now only called from `self._transform` which makes sure that there are no `np.NaN`s, right?

Reply:
Yeah, so that should be removed in the `_transform()` function, so we can check it inside the cpdef function. The idea is that the `transform_vector()` and `transform_scalar()` can also be called without calling `_transform()` first, i.e. when you can already guarantee yourself that the input is a scalar or a vector.

Reply:
> Furthermore, then you can call transform_vector() and transform_scalar() without calling _transform() first, i.e. when you can already guarantee yourself that the input is a scalar or a vector.

Yep, that'll be a great way to improve the speed of sampling and/or neighborhood generation. I'm really looking forward to that.